collector/exporter.go

```diff
@@ -165,226 +177,326 @@ func (e *Exporter) scrape(ch chan<- prometheus.Metric) {
 		e.error.Set(1)
 		return
 	}
-	isUpRows.Close()
 
+	isUpRows.Close()
 	e.mysqldUp.Set(1)
 
+	scrapeTime := time.Now()
+
 	if e.collect.SlowLogFilter {
-		sessionSettingsRows, err := db.Query(sessionSettingsQuery)
-		if err != nil {
-			log.Errorln("Error setting log_slow_filter:", err)
-			e.error.Set(1)
-			return
-		}
-		sessionSettingsRows.Close()
+		wg.Add(1)
+		go func() {
+			defer func() {
+				ch <- prometheus.MustNewConstMetric(scrapeDurationDesc, prometheus.GaugeValue, time.Since(scrapeTime).Seconds(), "connection")
+			}()
+			defer wg.Done()
+			sessionSettingsRows, err := db.Query(sessionSettingsQuery)
```

Comment:
This doesn't work. This query just changes a setting in the session, which won

Reply:
Most probably the session settings apply to the entire client object and all the connections in the connection pool. So, we don't need to do that for each and every connection. This is because the db client object hides the details about when new connections are created and when existing connections in the connection pool closed. 

But the session settings should have been applied once when we created the object, rather than doing that for every scrape. I will change that.

Reply:
The database/sql package and the drivers don't normally track any such changes and thus they will only apply to the current connection. This is ok with only 1 connection, but breaks with multiple connections. 

For these cases you'd normally use transactions since they guarantee that everything is executed on the same connection. But since we are changing a setting that should apply to all connections it should really be set on the pool itself (as explained in #254)